airbyte-integrations/connectors/destination-selectdb/src/main/java/io/airbyte/integrations/destination/selectdb/SelectdbDestination.java

```diff
@@ -0,0 +1,170 @@
+import io.airbyte.protocol.models.v0.DestinationSyncMode;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SelectdbDestination extends BaseConnector implements Destination {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SelectdbDestination.class);
+
+  private static final StandardNameTransformer namingResolver = new StandardNameTransformer();
+  private static Connection conn = null;
+  private static HttpUtil http = new HttpUtil();
+  static final String DESTINATION_TEMP_PATH_FIELD = "destination_temp_path";
+  private static final String JDBC_DRIVER = "com.mysql.cj.jdbc.Driver";
+  private static final String DB_URL_PATTERN = "jdbc:mysql://%s/%s?rewriteBatchedStatements=true&useUnicode=true&characterEncoding=utf8";
+
+
+
+  public static void main(String[] args) throws Exception {
+    new IntegrationRunner(new SelectdbDestination()).run(args);
+  }
+
+  @Override
+  public AirbyteConnectionStatus check(JsonNode config) {
+    try {
+      Preconditions.checkNotNull(config);
+      FileUtils.forceMkdir(getTempPathDir(config).toFile());
+      checkSelectdbAndConnect(config);
+    } catch (final Exception e) {
+      return new AirbyteConnectionStatus().withStatus(Status.FAILED).withMessage(e.getMessage());
+    }
+    return new AirbyteConnectionStatus().withStatus(Status.SUCCEEDED);
+  }
+
+
+  @Override
+  public AirbyteMessageConsumer getConsumer(JsonNode config,
+                                            ConfiguredAirbyteCatalog configuredCatalog,
+                                            Consumer<AirbyteMessage> outputRecordCollector) throws IOException, SQLException {
+    final Map<String, SelectdbWriteConfig> writeConfigs = new HashMap<>();
+    try {
+      final Path destinationDir = getTempPathDir(config);
+      FileUtils.forceMkdir(destinationDir.toFile());
+      for (ConfiguredAirbyteStream stream : configuredCatalog.getStreams()) {
+
+        final DestinationSyncMode syncMode = stream.getDestinationSyncMode();
+        if (syncMode == null) {
+          throw new IllegalStateException("Undefined destination sync mode");
+        }
+
+        final String streamName = stream.getStream().getName();
+        final String tableName = namingResolver.getIdentifier(streamName);
+        final String tmpTableName = namingResolver.getTmpTableName(streamName);
+        final Path tmpPath = destinationDir.resolve(tmpTableName + ".csv");
+        if (conn == null)
+          checkSelectdbAndConnect(config);
```

Comment:
why are you checking the connection here? that has already been done on check()

Reply:
done